tuf/repository_lib.py

```diff
@@ -1392,6 +1396,16 @@ def generate_snapshot_metadata(metadata_directory, version, expiration_date,
       The name of the repository.  If not supplied, 'rolename' is added to the
       'default' repository.
 
+    use_length:
+      Used to decide if the length attribute should be used. From version 1.0.1
```

Comment:
This implies that this is only true from version 1.0.1 of the spec, which I don't think is true? I'm not convinced the version, even if correct, adds value here either.

Reply:
It's not true. Even in tuf spec `1.0.0` for length in hashes in the snapshot metadata file it's written:
```
LENGTH is the integer length in bytes of the metadata file. It is OPTIONAL for all roles.

HASHES is the dictionary that specifies one or more hashes, including the cryptographic hash function. For example: { "sha256": HASH, ... }. It is OPTIONAL for all roles.
```
I thought that length and hashes were made optional as of version 1.0.1 but I was wrong.
Good catch!